src/runtime/plugins/documentDriven.ts

```diff
@@ -224,6 +224,7 @@ export default defineNuxtPlugin((nuxt) => {
 
       // Use `redirect` key to redirect to another page
       if (_page?.redirect) { return _page?.redirect }
+      if (_page?._dir?.navigation?.redirect) { return _page?._dir?.navigation?.redirect }
```

Comment:
I think we can leverage `navKeyFromPath(_page._path, 'redirect', navigation)` instead of fetching the `_dir.yml`